RELEASE.md

```diff
@@ -0,0 +1,56 @@
+Release type: minor
+
+Support for schema-extensions in subscriptions.
+
+i.e:
+```python
+from typing import AsyncIterator, List
+
+import strawberry
+from strawberry.extensions.base_extension import SchemaExtension
+
+
+@strawberry.type
+class Query:
+    @strawberry.field
+    def hello(self) -> str:
+        return "Hello, world!"
+
+
+@strawberry.type
+class Subscription:
+    @strawberry.subscription
+    async def notifications(self, info: strawberry.Info) -> AsyncIterator[str]:
+        async for notification in info.context["pubsub"].subscribe(
+            "notifications", info.context["user"]
+        ):
+            yield notification
+
+
+class MyExtension(SchemaExtension):
+
+    async def on_operation(self):
+        # This would run when the subscription starts
+        self.monitor_start(self.execution_context.operation_name)
+        yield
+        # The subscription has ended
+        self.monitor_end(self.execution_context.operation_name)
+
+    async def on_execute(self):
+        count = 0
+        # The subsctription is trying to yield a new result
+        self.monitor_start(
+            self.execution_context.operation_name, f"before yield {count}"
+        )
+        yield
+        # the subscription has yielded a new result
+        self.monitor_end(self.execution_context.operation_name, f"after yield {count}")
+        count += 1
+
+    # other hooks can are the same as per normal execution.
+
+
+schema = strawberry.Schema(
+    query=Query, subscription=Subscription, extensions=[MyExtension()]
```

Comment:
As of right now this code would not work, because the extension can only be initialized with a `execution_context` argument. This would work:

```suggestion
    query=Query, subscription=Subscription, extensions=[MyExtension]
```